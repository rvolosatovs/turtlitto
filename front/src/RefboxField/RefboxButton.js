import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

/**
 * A button part of the refbox. Does a callback to its parent to handle onClick events
 * Author: S.A. Tanja
 * Author: G.W. van der Heijden
 *
 * Props:
 *  - teamColor: the team color of the button (cyan or magenta).
 *  - onClick: a function on what to do when the button is pressed
 *  - tag: the tag of the button according to the TRC.
 */
const RefboxButton = props => {
  return (
    <Button
      teamColor={props.teamColor}
      onClick={() => {
        props.onClick();
      }}
    >
      {props.tag}
    </Button>
  );
};

/**
 * A styled component defining the style of the refbox buttons
 */
const Button = styled.button`
  background-color: ${props => props.teamColor};
  width: 20.6rem;
  height: 20.6rem;
  border: 0.25rem solid;
  font-size: 10rem;
`;

RefboxButton.propType = {
  teamColor: PropTypes.oneOf(["cyan", "magenta"]).isRequired,
  tag: PropTypes.oneOf(["KO", "FK", "GK", "TI", "C", "P"]).isRequired
};

export default RefboxButton;
