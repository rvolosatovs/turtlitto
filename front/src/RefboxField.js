import React, { Component } from "react";
import RefboxButton from "./RefboxButton";
import styled from "styled-components";

/**
 * Constructs a refbox for a team consisting of 6 buttons:
 * - Kick off (KO)
 * - Free kick (FK)
 * - Goal kick (GK)
 * - Throw in (TI)
 * - Corner (C)
 * - Penalty (P)
 * Also takes care of handeling commands that the buttons represents
 *
 * Parameter: teamColor, the color of the buttons inside of the refbox.
 *
 * Author: S.A. Tanja
 */
class RefboxField extends Component {
  constructor(props) {
    super(props);
    this.state = {
      teamColor: props.teamColor
    };
    this.HandleCommand = this.HandleCommand.bind(this);
  }

  render() {
    return (
      <div>
        <RefboxStyle>
          <RefboxButton
            color={this.state.teamColor}
            onClick={this.HandleCommand}
          >
            KO
          </RefboxButton>
          <RefboxButton
            color={this.state.teamColor}
            onClick={this.HandleCommand}
          >
            FK
          </RefboxButton>
          <RefboxButton
            color={this.state.teamColor}
            onClick={this.HandleCommand}
          >
            GK
          </RefboxButton>
          <RefboxButton
            color={this.state.teamColor}
            onClick={this.HandleCommand}
          >
            TI
          </RefboxButton>
          <RefboxButton
            color={this.state.teamColor}
            onClick={this.HandleCommand}
          >
            C
          </RefboxButton>
          <RefboxButton
            color={this.state.teamColor}
            onClick={this.HandleCommand}
          >
            P
          </RefboxButton>
        </RefboxStyle>
      </div>
    );
  }

  HandleCommand(color, command) {
    console.log("Button " + command + " was pressed from team " + color);
  }
}

/**
 * Places the buttons in 2 columns and rows of 3
 * Allows for multiple refboxes to be next to each other (float : left)
 */
const RefboxStyle = styled.div`
  display: grid;
  grid-template-columns: repeat(2, 206px);
  grid-template-rows: repeat(3, 206px);
  margin: auto;
  width: 412px;
  float: center;
`;

export default RefboxField;
