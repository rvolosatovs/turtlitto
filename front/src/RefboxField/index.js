import React from "react";
import RefboxButton from "./RefboxButton";
import styled from "styled-components";
import PropTypes from "prop-types";

/**
 * Constructs a refbox for a team consisting of 6 buttons:
 * - Kick off (KO)
 * - Free kick (FK)
 * - Goal kick (GK)
 * - Throw in (TI)
 * - Corner (C)
 * - Penalty (P)
 * Also takes care of handling commands that the buttons represents
 * Author: S.A. Tanja
 * Author: G.W. van der Heijden
 *
 * Props:
 *  - teamColor: the team color (cyan or magenta)
 *  - onClick: a function on what to do when the button is pressed
 */
const RefboxField = props => {
  return (
    <div>
      <Refbox>
        {["KO", "FK", "GK", "TI", "C", "P"].map(tag => {
          return (
            <RefboxButton
              key={tag}
              teamColor={props.teamColor}
              onClick={() => {
                props.onClick(tag, props.teamColor);
              }}
              tag={tag}
            />
          );
        })}
      </Refbox>
    </div>
  );
};

/**
 * Places the buttons in 2 columns and rows of 3
 * Allows for multiple refboxes to be next to each other (float : left)
 */
const Refbox = styled.div`
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  margin: auto;
  width: 41.2rem;
  float: center;
`;

RefboxField.propType = {
  teamColor: PropTypes.oneOf(["cyan", "magenta"]).isRequired
};

export default RefboxField;
