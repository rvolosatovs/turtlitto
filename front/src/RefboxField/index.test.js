import React from "react";
import RefboxField from ".";
import renderer from "react-test-renderer";

describe("RefboxField", () => {
  it("shows a cyan refbox field with all buttons", () => {
    const component = renderer.create(
      <RefboxField teamColor="cyan" onClick={(tag, color) => {}} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("shows a magenta refbox field with all buttons", () => {
    const component = renderer.create(
      <RefboxField teamColor="magenta" onClick={(tag, color) => {}} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
