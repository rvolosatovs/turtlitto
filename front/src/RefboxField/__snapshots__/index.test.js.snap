// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RefboxField shows a cyan refbox field with all buttons 1`] = `
<div>
  <div
    className="sc-bwzfXH dHSmib"
  >
    <button
      className="sc-bdVaJa jBNwEU"
      onClick={[Function]}
    >
      KO
    </button>
    <button
      className="sc-bdVaJa jBNwEU"
      onClick={[Function]}
    >
      FK
    </button>
    <button
      className="sc-bdVaJa jBNwEU"
      onClick={[Function]}
    >
      GK
    </button>
    <button
      className="sc-bdVaJa jBNwEU"
      onClick={[Function]}
    >
      TI
    </button>
    <button
      className="sc-bdVaJa jBNwEU"
      onClick={[Function]}
    >
      C
    </button>
    <button
      className="sc-bdVaJa jBNwEU"
      onClick={[Function]}
    >
      P
    </button>
  </div>
</div>
`;

exports[`RefboxField shows a magenta refbox field with all buttons 1`] = `
<div>
  <div
    className="sc-bwzfXH dHSmib"
  >
    <button
      className="sc-bdVaJa iwQbdo"
      onClick={[Function]}
    >
      KO
    </button>
    <button
      className="sc-bdVaJa iwQbdo"
      onClick={[Function]}
    >
      FK
    </button>
    <button
      className="sc-bdVaJa iwQbdo"
      onClick={[Function]}
    >
      GK
    </button>
    <button
      className="sc-bdVaJa iwQbdo"
      onClick={[Function]}
    >
      TI
    </button>
    <button
      className="sc-bdVaJa iwQbdo"
      onClick={[Function]}
    >
      C
    </button>
    <button
      className="sc-bdVaJa iwQbdo"
      onClick={[Function]}
    >
      P
    </button>
  </div>
</div>
`;
